src/components/material/periodical/helper.ts

```diff
@@ -28,4 +40,71 @@ export function makePeriodicalEditionsFromHoldings(holdings: HoldingsV3[]) {
     .flat();
 }
 
+export function filterAndSortPeriodicalEditions(baseData: {
+  [key: string]: PartialPeriodicalEdition[];
+}) {
+  const periodicalEditions = Object.entries(baseData);
+  const filteredEditions = periodicalEditions.map((yearAndEditions) => {
+    return yearAndEditions[1].reduce((acc, edition) => {
+      if (!edition.volumeNumber) {
+        return acc;
+      }
+      const includesValueAlready = acc.includes(edition.volumeNumber);
+      if (!includesValueAlready) {
+        acc.push(edition.volumeNumber);
+      }
+      return acc;
+    }, [] as string[]);
+  });
+  const allYears = periodicalEditions.map(
+    (yearEditionPair) => yearEditionPair[0]
+  );
+  const filteredPeriodicalEditionsObj = allYears.reduce((acc, curr, index) => {
+    // Sort editions array
+    // eslint-disable-next-line no-param-reassign
+    acc[curr] = filteredEditions[index].sort((a, b) => {
+      return a.localeCompare(b, "da-DK", { numeric: true });
+    });
+    return acc;
+  }, {} as { [key: string]: string[] });
+  return filteredPeriodicalEditionsObj;
+}
+
+export function handleSelectYear(
+  event: React.ChangeEvent<HTMLSelectElement>,
+  setYear: (value: React.SetStateAction<string>) => void,
+  selectPeriodicalHandler: (selectedPeriodical: PeriodicalEdition) => void,
+  periodicalEditions: { [key: string]: string[] },
+  groupList: GroupList
+) {
+  setYear(event.target.value);
+  // Updates the selectedPeriodical to the first edition of the selected year.
+  const changedEdition = getFirstEditionFromYear(
+    event.target.value,
+    periodicalEditions
+  );
+  const changedFullPeriodicalEdition = groupList[event.target.value].find(
+    (edition) => {
+      return edition.volumeNumber === changedEdition;
+    }
+  );
+  if (changedFullPeriodicalEdition) {
+    selectPeriodicalHandler(changedFullPeriodicalEdition);
+  }
+}
+
+export function handleSelectEdition(
+  event: React.ChangeEvent<HTMLSelectElement>,
+  groupList: GroupList,
+  year: string,
+  selectPeriodicalHandler: (selectedPeriodical: PeriodicalEdition) => void
+) {
+  const changedFullPeriodicalEdition = groupList[year].find((edition) => {
+    return edition.volumeNumber === event.target.value;
+  });
+  if (changedFullPeriodicalEdition) {
+    selectPeriodicalHandler(changedFullPeriodicalEdition);
+  }
+}
```

Comment:
Consider updating `handleSelectEdition()` to accept a volumeNumber instead of an event.

That would mean that  we could call it `handleSelectYear()` instead of having the code duplicated. It would also abstract our helpers from React events tied closely to the UI. 

handleSelectEdition

Reply:
Good point! I didn't see that one! Thanks 😉